lldb/include/lldb/Utility/CompletionRequest.h

```diff
@@ -157,6 +162,23 @@ class CompletionRequest {
 
   size_t GetCursorIndex() const { return m_cursor_index; }
 
+  size_t GetMaxReturnElements() const { return m_max_return_elements; }
+
+  /// Returns true if the maximum number of completions has not been reached
+  /// yet, hence we should keep adding completions.
+  bool ShouldAddCompletions() const {
+    return m_result.GetNumberOfResults() < m_max_return_elements;
+  }
+
+  /// Returns the maximum number of completions that need to be added
+  /// until reaching the maximum
+  size_t GetMaxNumberOfResultsToAdd() const {
+    const size_t number_of_results = m_result.GetNumberOfResults();
+    if (number_of_results >= m_max_return_elements)
```

Comment:
Shouldn't this just call `ShouldAddCompletions()` and return 0 if false?

Reply:
But still it needs to return the number of completions if `ShouldAddCompletions` is true

I Changed `ShouldAddCompletions` to return `GetMaxNumberOfCompletionsToAdd > 0` instead in order to reuse code